### PR TITLE
fix: check index for next word and replace global index with local one

### PR DESCRIPTION
**Pull request by hezhizhen**
## Problem
 

`nextWord()` used a global index and simply got the word from the list based on the index without checking if it's in the range.

## Solution
 

Pass the index as an argument to the function and let the callers maintain the index themselves (increase it here), and check if the index is valid.

I also fixed some typo and comments by the way.

### CODE REVIEW

**Comment by hezhizhen** [comment]
@skrdgraph

File: contrib/integration/swap/main.go [hunk 1]
```diff
@@ -137,10 +137,21 @@ func main() {
 
 }
 
+func getNextWord(index int) string {
+	// check if index is in the range of words
+	if index < 0 || index >= len(words) {
+		x.Log(errors.Errorf("invalid index: %d", index), "get next word")
```
Comment (mangalaman93) [inline contrib/integration/swap/main.go:143]: we should crash or something here instead of silently failing

**Review by mangalaman93 — CHANGES_REQUESTED** [review]
one comment, LG otherwise. Thanks for the PR

File: contrib/integration/swap/main.go (hunk 1 above)
Comment (hezhizhen) [reply contrib/integration/swap/main.go]: Done. PTAL

**Comment by mangalaman93** [comment]
rerunning the failed test. It is failing due to a known issue, we are looking into it.

**Review by mangalaman93 — APPROVED** [review]

**Comment by hezhizhen** [comment]
> rerunning the failed test. It is failing due to a known issue, we are looking into it.

@mangalaman93 How can I re-run failed tests?

**Review by MichelDiz — APPROVED** [review]

**Pull request merged**